libraries/standard/http/test/http_config.h

```diff
@@ -0,0 +1,25 @@
+#ifndef HTTP_CONFIG_H__
```

Comment:
I feel like we end up having to declare the same `http_config.h` and `mqtt_config.h` files multiple times. Cmake allows you to have a templated `http_config.h.in` or `mqtt_config.h.in`. By calling [configure_file(...)](https://cmake.org/cmake/help/v3.2/command/configure_file.html), we can update these config files without having to rewrite all of them ourselves.

Reply:
The point isn't to auto-populate the config files through. Within the demos (and tests), they are meant to explicitly exist for customer visibility to convey to them the way of configuring each library. Also, we are wanting to move away from an aggregate header file (config.h) to library-specific config files to avoid mixing all configurations in a single header file. (An aggregate header file can have unintended consequences on libraries that include the single file)